src/codegen/doc/mod.rs

```diff
@@ -0,0 +1,296 @@
+        ty.name = format!("{}Ext", class.convert().name);
+
+        for function in class.functions().iter() {
+            if function.parameters.iter().any(|p| p.instance_parameter) {
+                try!(create_fn_doc(w, &function, Some(Box::new(ty.clone()))));
+            }
+        }
+    }
+    Ok(())
+}
+
+fn create_enum_doc(w: &mut Write, enum_: &Enumeration,
+                   parent: Option<Box<TypeStruct>>) -> Result<()> {
+    let indent = compute_indent(&parent);
+    let mut ty = enum_.convert();
+    ty.parent = parent;
+    let tabs : String = primitives::tabs(indent);
+
+    if enum_.doc().is_some() || enum_.doc_deprecated().is_some() {
+        try!(writeln!(w, "{}{}", MOD_COMMENT, ty));
+    }
+
+    if let Some(ref enum_doc) = enum_.doc() {
+        try!(write_lines(w, &enum_doc, &tabs));
+    }
+    if let Some(ref enum_doc) = enum_.doc_deprecated() {
+        try!(write_deprecated(w, &tabs));
+        try!(write_lines(w, &enum_doc, &tabs));
+    }
+
+    let tabs : String = primitives::tabs(indent + 1);
+    for member_doc in enum_.members.iter() {
+        let mut sub_ty : TypeStruct = member_doc.convert();
+
+        if member_doc.doc().is_some() || member_doc.doc_deprecated().is_some() {
+            sub_ty.parent = Some(Box::new(ty.clone()));
+            try!(writeln!(w, "{}{}", MOD_COMMENT, sub_ty));
+        }
+        if let Some(ref m_doc) = member_doc.doc() {
+            try!(write_lines(w, &m_doc, &tabs));
+        }
+        if let Some(ref m_doc) = member_doc.doc_deprecated() {
+            try!(write_deprecated(w, &tabs));
+            try!(write_lines(w, &m_doc, &tabs));
+        }
+    }
+    Ok(())
+}
+
+fn create_fn_doc(w: &mut Write, fn_: &Function, parent: Option<Box<TypeStruct>>) -> Result<()> {
+    let tabs : String = primitives::tabs(compute_indent(&parent) + 1);
+
+    if fn_.doc().is_none() && fn_.doc_deprecated().is_none() && fn_.ret.doc().is_none() &&
+       fn_.ret.doc_deprecated().is_none() {
+        if fn_.parameters.iter().all(|x| {
+            x.doc().is_none() && x.doc_deprecated().is_none()
+        }) {
+            return Ok(());
+        }
+    }
+    let mut sub_ty = fn_.convert();
+    sub_ty.parent = parent;
+    try!(writeln!(w, "{}{}", MOD_COMMENT, sub_ty));
+
+    if let Some(ref docs) = fn_.doc() {
+        try!(write_lines(w, &docs, &tabs));
+    };
+    if let Some(ref docs) = fn_.doc_deprecated() {
+        try!(write_deprecated(w, &tabs));
+        try!(write_lines(w, &docs, &tabs));
+    };
+
+    if fn_.parameters.iter().any(|x| {
+           x.doc().is_some() || x.doc_deprecated().is_some()
+       }) {
+        try!(writeln!(w, "{}///", tabs));
+        try!(writeln!(w, "{}/// # Parameters", tabs));
+        try!(writeln!(w, "{}///", tabs));
+    }
+    let mut wrote = false;
+    for parameter in fn_.parameters.iter() {
+        if let Some(ref parameter_doc) = parameter.doc() {
+            try!(writeln!(w, "{}/// ## {}:", tabs, parameter.name));
```

Comment:
This is where `instance_parameter` is relevant too. Most likely it should be skipped because in Rust it's passed implicitly.